src/coreComponents/python/modules/geosx_xml_tools_package/geosx_xml_tools/tests/test_manager.py

```diff
@@ -115,35 +67,20 @@ def setUpClass(cls):
         cls.regexHandler.target['foo'] = '1.23'
         cls.regexHandler.target['bar'] = '4.56e7'
 
-    def evaluateRegex(self, parameterInput, expectedValue):
-        result = re.sub(regex_tools.patterns['parameters'], self.regexHandler, parameterInput)
-        self.assertEqual(result, expectedValue)
-
-    def test_parameter_regex_a(self):
-        self.evaluateRegex('$:foo*1.234', '1.23*1.234')
-
-    def test_parameter_regex_b(self):
-        self.evaluateRegex('$:foo*1.234/$:bar', '1.23*1.234/4.56e7')
-
-    def test_parameter_regex_c(self):
-        self.evaluateRegex('$:foo*1.234/($:bar*$:foo)', '1.23*1.234/(4.56e7*1.23)')
-
-    def test_parameter_regex_d(self):
-        self.evaluateRegex('$foo*1.234/$bar', '1.23*1.234/4.56e7')
-
-    def test_parameter_regex_e(self):
-        self.evaluateRegex('$foo$*1.234/$bar', '1.23*1.234/4.56e7')
-
-    def test_parameter_regex_f(self):
-        self.evaluateRegex('$foo$*1.234/$bar$', '1.23*1.234/4.56e7')
-
-    @unittest.expectedFailure
-    def test_parameter_regex_g(self):
-        self.evaluateRegex('$blah$*1.234/$bar$', '1.23*1.234/4.56e7')
-
-    @unittest.expectedFailure
-    def test_parameter_regex_h(self):
-        self.evaluateRegex('$foo$*1.234/$bar$', '4.56e7*1.234/4.56e7')
+    @parameterized.expand([['parameter_regex', '$:foo*1.234', '1.23*1.234'],
+                           ['parameter_regex', '$:foo*1.234/$:bar', '1.23*1.234/4.56e7'],
+                           ['parameter_regex', '$:foo*1.234/($:bar*$:foo)', '1.23*1.234/(4.56e7*1.23)'],
+                           ['parameter_regex', '$foo*1.234/$bar', '1.23*1.234/4.56e7'],
+                           ['parameter_regex', '$foo$*1.234/$bar', '1.23*1.234/4.56e7'],
+                           ['parameter_regex', '$foo$*1.234/$bar$', '1.23*1.234/4.56e7'],
+                           ['parameter_regex', '$blah$*1.234/$bar$', '1.23*1.234/4.56e7', True],
+                           ['parameter_regex', '$foo$*1.234/$bar$', '4.56e7*1.234/4.56e7', True]])
+    def test_sequence(self, name, parameterInput, expectedValue, expect_fail=False):
```

Comment:
`name` being unused here (and in other test member functions), I would simply rename `test_sequence` into `test_parameter_regex`. That would break a little of symmetry; but I really do not think it's important here.
It's always disturbing to have a parameter which is not used.

Reply:
The paremeterize decorator takes the first input and adds it to the test name, so the example I was using set it manually.  Removing this in all cases, works just fine.  It can create some odd looking names when I throw the regexes at them (e.g. test_symbolic_regex_06__1_0_2_0_2_), but they are numbered anyway so it doesn't matter.

Reply:
What's critical is that you are able to get the problematic parameters if it fails (which I'm quite sure should be OK).
If it's a common practice in this fwk, it's OK. 😉 Up to you!